apple/RNGestureHandlerButton.mm

```diff
@@ -64,10 +64,24 @@ - (BOOL)shouldHandleTouch:(RNGHUIView *)view
     return button.userEnabled;
   }
 
+  // Certain subviews such as RCTViewComponentView have been observed to have disabled
+  // accessibility gesture recognizers such as _UIAccessibilityHUDGateGestureRecognizer,
+  // ostensibly set by iOS. Such gesture recognizers cause us to return early, despite
+  // the returned view not actually responding to touches. This in turn breaks button
+  // touch handling. Therefore, as a bandaid we can check to ensure we don't return
+  // true here if the only gesture recognizers we have are all disabled.
+  BOOL hasEnabledGestureRecognizer = false;
+  for (UIGestureRecognizer *gestureRecognizer in view.gestureRecognizers) {
```

Comment:
I'm pretty sure that this will break on `macos` 😅

Reply:
Should I go ahead and move this implementation into `#if !TARGET_OS_OSX` for the sake of this fix, which may be an iOS-specific issue to begin with?

Reply:
Actually I would prefer something like this:

```obj-c
#if !TARGET_OS_OSX
  for (UIGestureRecognizer *gestureRecognizer in view.gestureRecognizers) {
#else
  for (NSGestureRecognizer *gestureRecognizer in view.gestureRecognizers) {
#endif
```

Now when I checked this `macos` did compile without issues, but I'm not sure why it does recognize `UIGestureRecognizer` if it has its own `NSGestureRecognizer`, so I believe it is better to split it into two parts. Also making it `iOS` exclusive may not be the best as it may also occur on `macos` (but for that we would need a reproduction)

Reply:
Got it, I'll take a look at such an implementation tomorrow

Reply:
I actually went with an implementation using `NSPredicate` to filter on `isEnabled == YES`, which I believe should work equally well on OSX as iOS? I don't have a good test setup to validate on OSX though, so please take a look and let me know if it works alright.

I also updated the comment, and confirmed that my snippet indeed produces the bug on 2.22.0 (Gestures' Text was unfortunately a non-starter), and that this proposed change addresses the bug using Xcode's debugger, so I think this PR should be good to go! Requesting re-review